### DISCUSSION TITLE
I want to know if the splitter supports custom separators

### DISCUSSION THREAD

**Discussion by Havedream6** [langchain4j/langchain4j#1485] [Q&A]
I want to know if the splitter supports custom separators

**Comment by PrimosK** [comment]
Hi Havedream6,

I guess you could use [DocumentByRegexSplitter.java](https://github.com/langchain4j/langchain4j/blob/main/langchain4j/src/main/java/dev/langchain4j/data/document/splitter/DocumentByRegexSplitter.java).

See also [DocumentByRegexSplitterTest.java](https://github.com/langchain4j/langchain4j/blob/d43a91f06eb158075d5412ea0cc50e40c6982499/langchain4j/src/test/java/dev/langchain4j/data/document/splitter/DocumentByRegexSplitterTest.java) for examples and tests of its usage..
